### PR TITLE
feat: Add Bitget exchange

### PR DESCRIPTION
**Pull request by nategurian**
- Added Bitget crypto exchange

### CODE REVIEW

File: src/xrc/src/exchanges.rs [hunk 1]
```diff
@@ -483,6 +483,45 @@ impl IsExchange for CryptoCom {
...
 
+/// Bitget
+#[derive(Deserialize)]
+struct BitgetResponse {
+    data: Vec<(String, String, String, String, String, String, String)>,
+}
+
+impl IsExchange for Bitget {
+    fn get_base_url(&self) -> &str {
+        "https://api.bitget.com/api/v2/mix/market/history-candles?symbol=BASE_ASSETQUOTE_ASSET&granularity=1m&productType=usdt-futures&startTime=START_TIME&endTime=END_TIME"
```
Comment (dfinity-ryancroote) [inline src/xrc/src/exchanges.rs:494]: It looks like this is using the Bitget's futures API. I'm not sure if there is much difference between the two, but I think we want to use the spot API.

Comment (nategurian) [reply src/xrc/src/exchanges.rs]: Updated to the spot API

File: src/xrc-tests/src/tests/get_icp_xdr_rate.rs [hunk 2]
```diff
@@ -24,38 +24,30 @@ use crate::{
...
+///          2888501408, 2888501408, 2895907822, 2895907822, 2895907822, 2895907822, 2896648464, 2896648464, 2896648464, 2896648464, 2903314236, 2903314236, 2903314236, 2903314236, 2903314236, 2903314236, 2903314236, 2903314236, 2910720650, 2910720650, 2910720650, 2910720650, 2937481433, 2937481433, 2946854972, 2946854972, 2946854972, 2946854972, 2954411010, 2954411010, 2955166614, 2955166614, 2961967049, 2961967049, 2961967049, 2961967049, 2966268754, 2966268754, 2966268754, 2966268754, 2969523087, 2969523087, 3007915659, 3007915659, 3007915659, 3007915659, 3007915659, 3007915659, 3007915659, 3007915659, 3015628263, 3015628263, 3015628263, 3015628263, 3016399524, 3016399524, 3016399524, 3016399524, 3023340868, 3023340868, 3023340868, 3023340868, 3023340868, 3023340868, 3023340868, 3023340868, 3026193375, 3026193375, 3031053472, 3031053472, 3031053472, 3031053472, 3071240197, 3071240197, 3071240197, 3071240197, 3079115172, 3079115172, 3079902669, 3079902669, 3086990147, 3086990147, 3086990147, 3086990147, 3088898004, 3088898004, 3088898004, 3088898004, 3094865122, 3094865122, 3153927433, 3153927433 ]
```
Comment (dfinity-ryancroote) [inline src/xrc-tests/src/tests/get_icp_xdr_rate.rs:47]: Can we get some better formatting for this? 😄

Comment (nategurian) [reply src/xrc-tests/src/tests/get_icp_xdr_rate.rs:47]: I dont know how that slipped past me. I will fix it

**Review by THLO — APPROVED** [review]

**Review by dfinity-ryancroote — APPROVED** [review]

**Pull request merged**